src/transformers/models/tvlt/modeling_tvlt.py

```diff
@@ -0,0 +1,1380 @@
+
+        pixel_values_mixed (`torch.FloatTensor` of shape `(batch_size, num_frames, num_channels, height, width)`):
+            Pixel masks of pixel_values_mixed. Pixel values mixed can be obtained using [`TvltPixelFeatureExtractor`].
+            See [`TvltPixelFeatureExtractor.__call__`] for details.
+
+        pixel_masks_mixed (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values that mixe positive and negative samples in Tvlt vision-audio matching. Audio values can be
+            obtained using [`TvltPixelFeatureExtractor`]. See [`TvltPixelFeatureExtractor.__call__`] for details.
+
+        labels (`torch.FloatTensor` of shape `(batch_size, num_labls)`):
+            Number of labels for pretrainings. For example, whether vision and audio match in pretraining.
+
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    "The bare TVLT Model transformer outputting raw hidden-states without any specific head on top.",
+    TVLT_START_DOCSTRING,
+)
+class TvltModel(TvltPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.config = config
+
+        self.pixel_embeddings = TvltPixelEmbeddings(config)
+        self.audio_embeddings = TvltAudioEmbeddings(config)
+        self.encoder = TvltEncoder(config)
+
+        self.cls_embedding = nn.Parameter(torch.zeros(1, 1, config.hidden_size))
+
+        if config.use_mean_pooling:
+            self.layernorm = None
+        else:
+            self.layernorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.pixel_embeddings.patch_embeddings, self.audio_embeddings.patch_embeddings
+
+    def _prune_heads(self, heads_to_prune):
+        """
+        Prunes heads of the model. heads_to_prune: dict of {layer_num: list of heads to prune in this layer} See base
+        class PreTrainedModel
+        """
+        for layer, heads in heads_to_prune.items():
+            self.encoder.layer[layer].attention.prune_heads(heads)
+
+    @add_start_docstrings_to_model_forward(TVLT_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=TvltModelOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        pixel_values,
+        audio_values,
+        pixel_masks=None,
+        audio_masks=None,
+        pixel_mask_pos_perm=None,
```

Comment:
The TVLT_INPUTS_DOCSTRING doesn't seem to document the arguments here, like pixel_mask_pos_perm?

Reply:
Also, pixel_mask_pos_perm is not the bestname, do you mean pixel_mask_position_permutation?